firefox-extension/background.js

```diff
@@ -0,0 +1,40 @@
+const newTabIds = []
```

Comment:
This will stop working when manifest v3 is in place, as the script will be shut down and recreated when needed.
If you need to store something, use session storage: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/storage/session

Reply:
MV3 should not be a problem on Firefox correct?

Reply:
Firefox will also implement MV3. There are exceptions, but not related to this part.
See: https://blog.mozilla.org/addons/2022/06/08/manifest-v3-firefox-developer-preview-how-to-get-involved/